osu.Game/Rulesets/Scoring/HealthProcessor.cs

```diff
@@ -69,6 +69,28 @@ protected override void RevertResultInternal(JudgementResult result)
         /// </summary>
         protected virtual bool DefaultFailCondition => Precision.AlmostBigger(Health.MinValue, Health.Value);
 
+        /// <summary>
+        /// Whether the current state of <see cref="HealthProcessor"/> or the provided <paramref name="result"/> meets the fail conditions.
+        /// </summary>
+        /// <param name="result">The judgement result.</param>
+        private bool meetsFailConditions(JudgementResult result)
```

Comment:
Would rename to `meetsAnyFailCondition` to better reflect what this is actually doing. If named `meetsFailCondition`, it may suggest that all fail conditions are being checked in conjunction rather than as alternatives.